paddle/fluid/platform/device_context.cc

```diff
@@ -397,6 +397,10 @@ thread_local std::string cur_input_shape_str = "";
 // the cache capacity of different input shapes for MKLDNN.
 // Default 1 means fixed input shape, not dynamic shape.
 thread_local int cur_input_shape_cache_capacity = 1;
+// Recently registered data_format. This is needed to
+// know for converting MKL-DNN Tensor to non MKL-DNN
+thread_local paddle::framework::DataLayout cur_paddle_data_layout =
+    paddle::framework::DataLayout::kNCHW;
```

Comment:
@luotao1 Is it possible to get layout format from PaddleTensor class? if possible, I think we can combine layout info with cur_input_shape_str.

Reply:
@LeoZhao-Intel Do you mean here?
https://github.com/PaddlePaddle/Paddle/blob/d5ff79e55ea329396f4c61686a318b7c77520b93/paddle/fluid/framework/tensor.h#L37-L42
Do you have any idea on getting layout format from here?

Reply:
https://github.com/PaddlePaddle/Paddle/blob/d5ff79e55ea329396f4c61686a318b7c77520b93/paddle/fluid/inference/api/analysis_predictor.cc#L206

What I mean is if we can combine layout with input shape, and for other places, we use read layout by reusing same global variable.

But I am not sure if this change is for both inference and training, if for both, then my idea is useless.

Reply:
> But I am not sure if this change is for both inference and training

This PR is for training as well.

Reply:
Then not applicable. skip my comments here.

Reply:
@LeoZhao-Intel , @luotao1 Just one note here , in current design when *data_format=NHWC* then still Tensors layout is *kNCHW*. So:
1) I'm not sure what is current reading of Tensor::layout .
2) What does DataLayout::AnyLayout mean ?

Reply:
Honestly speaking, I am still a little confused here, we have 2 kinds of format, one is Paddle Tensor layout, the other is MKLDNN layout, how to map them together?

for input Paddle Tensor is kNHWC, what is real value reading from tensor?
what's MKLDNN input and output layout for different input Paddle Tensor layout?

can we  have a table to show clearly?